substreams/crates/tycho-substreams/src/block_storage.rs

```diff
@@ -0,0 +1,79 @@
+use std::collections::HashMap;
+
+use substreams_ethereum::pb::{eth, eth::v2::block::DetailLevel};
+
+use crate::{
+    models::{ContractSlot, StorageChanges, Transaction},
+    pb::tycho::evm::v1::TransactionStorageChanges,
+};
+
+#[allow(dead_code)]
+/// Helper function to extract all storage changes on a block.
+/// The raw block information collected is intended to be used by the DCI (Dynamic Contract Indexer)
+/// to extract and index relevant changes. This is specifically for dynamically identified contracts
+/// that the DCI has chosen to index. Note that core protocol data should still be properly
+/// integrated and indexed by the substreams package as per usual.
+///
+/// ## Panics
+/// Panics if the provided block is not an extended block model, as indicated by its detail level.
+///
+/// ## Warning
+/// ⚠️ This function *only* works if the **extended block model** is available,
+/// more [here](https://streamingfastio.medium.com/new-block-model-to-accelerate-chain-integration-9f65126e5425)
+fn get_block_storage_changes(block: &eth::v2::Block) -> Vec<TransactionStorageChanges> {
+    if block.detail_level != Into::<i32>::into(DetailLevel::DetaillevelExtended) {
+        panic!("Only extended blocks are supported");
+    }
+    let mut block_storage_changes = vec![];
```

Comment:
Optimisation: if you pre-allocate a capacity for something that can change size, you help avoid run-time re-allocations of memory as it grows. 
```suggestion
    let mut block_storage_changes = Vec::with_capacity(transactions.len());
```